src/shortcuts/redo.ts

```diff
@@ -1,18 +1,9 @@
 import RedoIcon from '../components/RedoIcon'
-import { Patch, Shortcut } from '../@types'
+import { Shortcut } from '../@types'
 import { isRedoEnabled } from '../selectors/isRedoEnabled'
 import { alert as alertAction } from '../action-creators'
-import { NAVIGATION_ACTIONS } from '../constants'
 import { startCase } from 'lodash'
-
-/**
- * Recursively calculates last action type from patches history if it is one of the navigation actions and finally returns the action.
- */
-const getLatestActionType = (patches: Patch[], n = 1): string => {
-  const lastActionType = patches[patches.length - n]?.[0]?.actions[0]
-  if (NAVIGATION_ACTIONS[lastActionType]) return getLatestActionType(patches, n + 1)
-  return lastActionType
-}
+import { getLatestActionType } from '../util/getLastActionType'
 
 const redoShortcut: Shortcut = {
```

Comment:
Could you change `isActive` to `canExecute` while we're at it? `isActive` only changes the highlighting, not the click state.

Reply:
Sure